pypesto/C.py

```diff
@@ -0,0 +1,128 @@
+"""
+Constants
+=========
+
+Package-wide consistent constant definitions.
+
+"""
+
+from enum import Enum
```

Comment:
how about turning all of these into proper enums? I think with multiple inheritance from `str` and `Enum` we should maintain backwards compatibility.

Would keep imports smaller and enable better input checking.

Reply:
What for example would you mean? Regarding brevity of imports, I had in mind anyhow only importing `from pypesto import C` and then using e.g. `C.MODE_FUN`, which should be nice and readable already.

Reply:
> <img alt="image" width="345" src="https://user-images.githubusercontent.com/14923969/147430847-201e57d9-c43d-4a58-b8fe-dfff3f59adb0.png">
> 
> can we keep things in separate (minimal) PRs if >100 files are touched? This is an absolute pain to review.

Yeah ... the C + refactoring of the ensemble module became necessary along the way, but will try to extract the actual code changes there, to separate from the mere sorting.